### ISSUE TITLE
Questions on using Dockerfile and Ubuntu

### ISSUE THREAD

**Issue by noyuzrad** [DBraun/DawDreamer#82]
Hi

I tried pip install but it seems that pip does recognize the package. 
Also tried to git clone and follow docker installation but failed with errors

Any advice?

**Comment by DBraun** [comment]
Hi, sorry the README was misleading about pip on Linux. I changed it now. Can you upload some of your log from using the dockerfile?

**Comment by noyuzrad** [comment]
Thanks for replying. your package is great. I currently run it Windows but I need to use it on my server which runs ubuntu.
As for the problem I get the below log.
I tried to search it on google but got do dead end

Sending build context to Docker daemon  739.3MB
Step 1/15 : FROM ubuntu:21.04 as ubuntu_base
 ---> 7cc39f89fa58
Step 2/15 : ARG DEBIAN_FRONTEND=noninteractive
 ---> Using cache
 ---> 93c58662358e
Step 3/15 : RUN apt-get update -yq && apt-get install -yq --no-install-recommends     ca-certificates     build-essential     clang     pkg-config     libboost-all-dev     libboost-python-dev     libfreetype6-dev     libx11-dev     libxinerama-dev     libxrandr-dev     libxcursor-dev     mesa-common-dev     libasound2-dev     freeglut3-dev     libxcomposite-dev     libcurl4-gnutls-dev     git     cmake     python3     python3.9-dev     faust     libsamplerate0     libsndfile1     llvm-11     llvm-11-dev && update-ca-certificates && apt-get clean -y
 ---> Running in 5c0de00836b6
Get:1 http://security.ubuntu.com/ubuntu hirsute-security InRelease [110 kB]
Get:2 http://archive.ubuntu.com/ubuntu hirsute InRelease [269 kB]
**Err:1 http://security.ubuntu.com/ubuntu hirsute-security InRelease
  gpgv, gpgv2 or gpgv1 required for verification, but neither seems installed
Get:3 http://archive.ubuntu.com/ubuntu hirsute-updates InRelease [115 kB]
Err:2 http://archive.ubuntu.com/ubuntu hirsute InRelease
  gpgv, gpgv2 or gpgv1 required for verification, but neither seems installed
Get:4 http://archive.ubuntu.com/ubuntu hirsute-backports InRelease [101 kB]
Err:3 http://archive.ubuntu.com/ubuntu hirsute-updates InRelease
  gpgv, gpgv2 or gpgv1 required for verification, but neither seems installed
Err:4 http://archive.ubuntu.com/ubuntu hirsute-backports InRelease
  gpgv, gpgv2 or gpgv1 required for verification, but neither seems installed
Reading package lists...
W: GPG error: http://security.ubuntu.com/ubuntu hirsute-security InRelease: gpgv, gpgv2 or gpgv1 required for verification, but neither seems installed
E: The repository 'http://security.ubuntu.com/ubuntu hirsute-security InRelease' is not signed.
W: GPG error: http://archive.ubuntu.com/ubuntu hirsute InRelease: gpgv, gpgv2 or gpgv1 required for verification, but neither seems installed
E: The repository 'http://archive.ubuntu.com/ubuntu hirsute InRelease' is not signed.
W: GPG error: http://archive.ubuntu.com/ubuntu hirsute-updates InRelease: gpgv, gpgv2 or gpgv1 required for verification, but neither seems installed
E: The repository 'http://archive.ubuntu.com/ubuntu hirsute-updates InRelease' is not signed.
W: GPG error: http://archive.ubuntu.com/ubuntu hirsute-backports InRelease: gpgv, gpgv2 or gpgv1 required for verification, but neither seems installed**
E: The repository 'http://archive.ubuntu.com/ubuntu hirsute-backports InRelease' is not signed.
The command '/bin/sh -c apt-get update -yq && apt-get install -yq --no-install-recommends     ca-certificates     build-essential     clang     pkg-config     libboost-all-dev     libboost-python-dev     libfreetype6-dev     libx11-dev     libxinerama-dev     libxrandr-dev     libxcursor-dev     mesa-common-dev     libasound2-dev     freeglut3-dev     libxcomposite-dev     libcurl4-gnutls-dev     git     cmake     python3     python3.9-dev     faust     libsamplerate0     libsndfile1     llvm-11     llvm-11-dev && update-ca-certificates && apt-get clean -y' returned a non-zero code: 100

**Comment by DBraun** [comment]
Can you try some of the suggestions here? https://stackoverflow.com/a/51752997

**Comment by noyuzrad** [comment]
This did not help

also see that from the error:
`W: GPG error: http://security.ubuntu.com/ubuntu hirsute-security InRelease: gpgv, gpgv2 or gpgv1 required for verification, but neither seems installed
E: The repository 'http://security.ubuntu.com/ubuntu hirsute-security InRelease' is not signed.
`

the required packages are gpgv not gnup as in your link. One way or another I installed them all using 
`sudo apt-get install -y gpgv1
`

still got failed..

**Comment by DBraun** [comment]
Is your version of docker recent? https://stackoverflow.com/a/70834272

**Comment by noyuzrad** [comment]
yes, I did come across this stackoverflow page as well.
running the commands yields

`Client:
 Version:           18.09.4
 API version:       1.39
 Go version:        go1.10.8
 Git commit:        d14af54266
 Built:             Wed Mar 27 18:35:44 2019
 OS/Arch:           linux/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          18.09.4
  API version:      1.39 (minimum version 1.12)
  Go version:       go1.10.8
  Git commit:       d14af54
  Built:            Wed Mar 27 18:01:48 2019
  OS/Arch:          linux/amd64
  Experimental:     false
`

**Comment by DBraun** [comment]
Maybe 18.09.4 is too low? This is the output from the [Dockerfile GitHub continuous integration](https://github.com/DBraun/DawDreamer/runs/5824821217?check_suite_focus=true):

```
2022-04-04T22:20:35.6241368Z Requested labels: ubuntu-latest
2022-04-04T22:20:35.6241415Z Job defined at: DBraun/DawDreamer/.github/workflows/all.yml@refs/tags/v0.6.1
2022-04-04T22:20:35.6241438Z Waiting for a runner to pick up this job...
2022-04-04T22:20:37.3538640Z Job is waiting for a hosted runner to come online.
2022-04-04T22:20:40.4808991Z Job is about to start running on the hosted runner: GitHub Actions 6 (hosted)
2022-04-04T22:20:44.3474742Z Current runner version: '2.289.2'
2022-04-04T22:20:44.3504176Z ##[group]Operating System
2022-04-04T22:20:44.3504720Z Ubuntu
2022-04-04T22:20:44.3505025Z 20.04.4
2022-04-04T22:20:44.3505319Z LTS
2022-04-04T22:20:44.3505595Z ##[endgroup]
2022-04-04T22:20:44.3506036Z ##[group]Virtual Environment
2022-04-04T22:20:44.3506452Z Environment: ubuntu-20.04
2022-04-04T22:20:44.3506766Z Version: 20220330.0
2022-04-04T22:20:44.3507336Z Included Software: https://github.com/actions/virtual-environments/blob/ubuntu20/20220330.0/images/linux/Ubuntu2004-Readme.md
2022-04-04T22:20:44.3508137Z Image Release: https://github.com/actions/virtual-environments/releases/tag/ubuntu20%2F20220330.0
2022-04-04T22:20:44.3508595Z ##[endgroup]
2022-04-04T22:20:44.3508988Z ##[group]Virtual Environment Provisioner
2022-04-04T22:20:44.3509386Z 1.0.0.0-main-20220307-1
2022-04-04T22:20:44.3509754Z ##[endgroup]
2022-04-04T22:20:44.3510783Z ##[group]GITHUB_TOKEN Permissions
2022-04-04T22:20:44.3511467Z Actions: write
2022-04-04T22:20:44.3511972Z Checks: write
2022-04-04T22:20:44.3512369Z Contents: write
2022-04-04T22:20:44.3512721Z Deployments: write
2022-04-04T22:20:44.3513082Z Discussions: write
2022-04-04T22:20:44.3513384Z Issues: write
2022-04-04T22:20:44.3513709Z Metadata: read
2022-04-04T22:20:44.3514081Z Packages: write
2022-04-04T22:20:44.3514375Z Pages: write
2022-04-04T22:20:44.3514715Z PullRequests: write
2022-04-04T22:20:44.3515108Z RepositoryProjects: write
2022-04-04T22:20:44.3515468Z SecurityEvents: write
2022-04-04T22:20:44.3515853Z Statuses: write
2022-04-04T22:20:44.3516196Z ##[endgroup]
2022-04-04T22:20:44.3519900Z Secret source: Actions
2022-04-04T22:20:44.3520482Z Prepare workflow directory
2022-04-04T22:20:44.4351298Z Prepare all required actions
2022-04-04T22:20:44.4534034Z Getting action download info
2022-04-04T22:20:44.6538806Z Download action repository 'actions/checkout@v2' (SHA:ec3a7ce113134d7a93b817d10a8272cb61118579)
2022-04-04T22:20:45.6050054Z Download action repository 'actions/setup-python@v2' (SHA:7f80679172b057fc5e90d70d197929d454754a5a)
```

It uses `20.04.4`, which is still lower than `21.04` but it is higher than `18.09.4`

**Comment by DBraun** [comment]
`import dawdreamer`
or
`import dawdreamer as daw`
are the suggested way.

With PyCharm you have to configure the Python interpreter.

If you installed dawdreamer with pip, you can do `pip --version` which should print the version and location of pip. For example, on Windows it's `pip 22.0.4 from C:\Python39\lib\site-packages\pip (python 3.9)` for me. I then double check that there's a dawdreamer folder in the site-packages folder.

**Comment by DBraun** [comment]
bumping because I updated my answer above.

**Comment by noyuzrad** [comment]
Still not working.

I attach my output for running the docker and trying to import.
As well, seems dawdreamer does not create its build in lib/site-packages of python3.9.
I tried also to use virtual environment in pycharm with python3.9, including global site-packages. DawDreamer is not imported.
I suspect that DawDreamer does not build at all, or installed to unknown destination.
Also maybe I don't make proper use of docker since I am new to it - should I activate some docker image/environment or something like that? 
![image](https://user-images.githubusercontent.com/57722241/162713872-6a8490a6-b4bd-4874-a2d3-c677f1d8a0dd.png)

**Comment by DBraun** [comment]
It looks like you've built the docker image, but now you need to run it.

You can either plainly run it
`docker run dawdreamer` and then `python my_script.py`

or configure PyCharm to use it
https://www.jetbrains.com/help/pycharm/using-docker-as-a-remote-interpreter.html#run

**Comment by noyuzrad** [comment]
Ok I managed to succeed! I had to dig inside docker as a first timer and it is not that simple. 
My steps were to run a modified version of your run command, in order to communicate with the container:
`docker run -it dawdreamer /bin/bash`

Also, in order to run my script using DawDreamer, I had to 
1. share volume with the container using a fuller instruction:
`docker run -v "<path_to_local_volume>":"<path_at _the_container>" -it dawdreamer /bin/bash`
2. Add my imported packages to your DockerFile where pip install is..
3. moreover, I had to place the dawdreamer.so file resided in the DawDreamer/tests folder into my own project root folder to be able to access it.

Phiuu that was quite a journey... 
Not sure If I did it the most elegant or correct way, but I did.

Thanks for all the ongoing support!
